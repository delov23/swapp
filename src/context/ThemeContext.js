import { createContext } from 'react';

export const LIGHT_THEME = 'light';
export const DARK_THEME = 'dark';

const ThemeContext = createContext(LIGHT_THEME);

export default ThemeContext;
