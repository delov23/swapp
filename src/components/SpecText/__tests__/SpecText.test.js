import React from 'react';
import { shallow, mount } from 'enzyme';

import { SpecText } from '../';
import ThemeContext, {
  LIGHT_THEME,
  DARK_THEME,
} from '../../../context/ThemeContext';

const EXAMPLE_PROPS = {
  left: 'Bonus points',
  right: 9000,
};

describe('<SpecText />', () => {
  it('should match snapshot', () => {
    expect(shallow(<SpecText {...EXAMPLE_PROPS} />)).toMatchSnapshot();
  });

  it('should render info from props', () => {
    const wrapper = mount(<SpecText {...EXAMPLE_PROPS} />);

    expect(wrapper.html()).toContain('Bonus points: ');
    expect(wrapper.html()).toContain('9000');
  });

  it('should support LIGHT_THEME', () => {
    const wrapper = mount(
      <ThemeContext.Provider value={LIGHT_THEME}>
        <SpecText {...EXAMPLE_PROPS} />
      </ThemeContext.Provider>,
    );

    expect(wrapper.find('.leftLight').text()).toEqual('Bonus points: ');
    expect(wrapper.find('.rightLight').text()).toEqual('9000');
  });

  it('should support DARK_THEME', () => {
    const wrapper = mount(
      <ThemeContext.Provider value={DARK_THEME}>
        <SpecText {...EXAMPLE_PROPS} />
      </ThemeContext.Provider>,
    );

    expect(wrapper.find('.leftDark').text()).toEqual('Bonus points: ');
    expect(wrapper.find('.rightDark').text()).toEqual('9000');
  });
});
