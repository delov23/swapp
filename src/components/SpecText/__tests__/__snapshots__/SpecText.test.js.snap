// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpecText /> should match snapshot 1`] = `
<p
  className="text"
>
  <span
    className="leftLight"
  >
    Bonus points
    : 
  </span>
  <span
    className="rightLight"
  >
    9000
  </span>
</p>
`;
